dash/testing/application_runners.py

```diff
@@ -214,3 +223,55 @@ def stop(self):
                 )
                 self.proc.kill()
                 self.proc.communicate()
+
+
+class RRunner(ProcessRunner):
+    def __init__(self, keep_open=False, stop_timeout=3):
+        super(RRunner, self).__init__(
+            keep_open=keep_open, stop_timeout=stop_timeout
+        )
+        self.proc = None
+
+    # pylint: disable=arguments-differ
+    def start(self, app):
+        """Start the server with waitress-serve in process flavor """
+
+        # app is a R string chunk
+        if not (os.path.isfile(app) and os.path.exists(app)):
+            path = (
+                "/tmp/app_{}.R".format(uuid.uuid4().hex)
+                if not self.is_windows
+                else os.path.join(
+                    (os.getenv("TEMP"), "app_{}.R".format(uuid.uuid4().hex))
+                )
+            )
```

Comment:
Seems like this will work, but it doesn't remove the file afterward, does it? Would there have been a way to do this with `tmpdir`? Not sure how to go about using one fixture from within the definition of another, but it feels like all the edge cases and cleanup have already been sorted out there.